### PR TITLE
Support python 3.7

### PR DESCRIPTION
**Pull request by minrk**
- bump jupyterlab to a version packaged on py37 (this is the main change that's required)
- add 3.7 to freeze list
- specify nbconvert 5.4 to avoid falling back to 5.3 in order to get pandoc 2.x (which doesn't work)

cc @allisons who reported Python 3.7 not working on the jupyterhub/binder gitter

### CODE REVIEW

File: repo2docker/buildpacks/conda/environment.frozen.yml [hunk 1]
```diff
@@ -36,40 +25,34 @@ dependencies:
...
   - markupsafe=1.0=py36h470a237_1
   - mistune=0.8.3=py36h470a237_2
-  - nbconvert=5.4.0=0
+  - nbconvert=5.4.0=1
   - nbformat=4.4.0=py_1
   - ncurses=6.1=hfc679d8_1
   - notebook=5.6.0=py36_1
   - openssl=1.0.2p=h470a237_0
-  - pandoc=2.2.2=1
+  - pandoc=1.19.2=0
```
[the remark this answers is not in the document]
Comment (minrk) [reply repo2docker/buildpacks/conda/environment.frozen.yml]: It shouldn't. There are compatibility problems with pandoc 2 and nbconvert, which is why this got pinned down.

File: repo2docker/buildpacks/conda/environment.frozen.yml [hunk 2]
```diff
@@ -79,15 +62,13 @@ dependencies:
...
   - traitlets=4.3.2=py36_0
-  - twisted=18.7.0=py36h470a237_1
   - wcwidth=0.1.7=py_1
   - webencodings=0.5.1=py_1
   - wheel=0.31.1=py36_1
   - widgetsnbextension=3.2.1=py36_1
   - xz=5.2.4=h470a237_1
   - zeromq=4.2.5=hfc679d8_5
   - zlib=1.2.11=h470a237_3
-  - zope.interface=4.5.0=py36h470a237_1
```
[the remark this answers is not in the document]
Comment (minrk) [reply repo2docker/buildpacks/conda/environment.frozen.yml]: It's not clear to me why these got pulled in, but they showed up in a [recent refreeze](https://github.com/jupyter/repo2docker/commit/710fdcb24eb64eef2878e97be0e20b4f9ad3e0cb). My guess is they got pulled in by some dependency that wasn't actually required. I don't think they should be present.

**Comment by jzf2101** [comment]
Are we ready to merge?

**Comment by minrk** [comment]
@jzf2101 I believe it's all set.

**Pull request merged**